kafka-client/src/main/java/dev/responsive/kafka/internal/clients/StoreCommitListener.java

```diff
@@ -27,14 +28,14 @@ private void onCommit(
     for (final var e : committedOffsets.entrySet()) {
       final TopicPartition p = e.getKey().getPartition();
       for (final ResponsiveStoreRegistration storeRegistration
-          : storeRegistry.getRegisteredStoresForChangelog(p)) {
+          : storeRegistry.getRegisteredStoresForChangelog(p, threadId)) {
```

Comment:
unrelated, but do you mind adding a comment about why we do this twice (once for committed offsets and once for written offsets)? it's not obviously clear to me what's the difference

Reply:
one path gets triggered for changelogs, the other for source topics (when the source table optimization is on)